client/jetpack-cloud/sections/agency-dashboard/sites-overview/sites-dataviews/index.tsx

```diff
@@ -23,7 +23,13 @@ const SitesDataViews = ( {
 }: SitesDataViewsProps ) => {
 	const translate = useTranslate();
 
-	const sites = useFormattedSites( data?.sites ?? [] );
+	const totalSites = data ? data.sites.length : 0;
+	const sitesPerPage = sitesViewState.perPage;
+	const totalPages = Math.ceil( totalSites / sitesPerPage );
+	const pageEndPoint = sitesViewState.page * sitesPerPage;
+	const pageStartPoint = pageEndPoint - ( sitesPerPage - 1 );
+	const unformattedSites = data ? data.sites.slice( pageStartPoint - 1, pageEndPoint ) : [];
+	const sites = useFormattedSites( unformattedSites );
```

Comment:
A few points:

* `totalSites` assumes `data.sites` exists. Previously we didn't have this assumption. Does it always exist?
* `sitesPerPage`: I'd suggest requiring sitesPerPage to be `>0` in case something is stored wrong on the backend. `0` would give a div-by-zero error, and negatives would throw all the math off.
* `unformattedSites` also has the assumption that `data.sites` exists.

As for `pageStartPoint`, you're adding 1 to its value, and then you subtract one on the only time you use it in the `slice`. Perhaps just set its value to `pageEndPoint - sitesPerPage`?

Given that arrays are zero-indexed, in their current states `pageStartPoint` is really "first site plus one", and `pageEndPoint` is really "next first site".

Reply:
Mainly agree with Thomas.
Didn't test, but some suggestions:
```suggestion
	const sitesPerPage = Math.max( 1, sitesViewState.perPage );
	const totalPages = Math.ceil( totalSites / sitesPerPage );
	const pageEndPoint = sitesViewState.page * sitesPerPage;
	const pageStartPoint = Math.max( 1, pageEndPoint - sitesPerPage + 1 );
	const unformattedSites = data?.sites?.slice( pageStartPoint - 1, pageEndPoint ) || [];
	const sites = useFormattedSites( unformattedSites );
```

Reply:
Great! Though I'd tweak it to this (untested):

```
	const pageStartPoint = pageEndPoint - sitesPerPage;
	const unformattedSites = data?.sites?.slice( pageStartPoint, pageEndPoint ) || [];
```

Reply:
In the end, only needed `totalPages`, `totalSites` and `sitesPerPage`.